packages/node-sdk/src/client.ts

```diff
@@ -1084,12 +1093,24 @@ export class BoundBucketClient {
 function checkContextWithTracking(context: ContextWithTracking) {
   ok(isObject(context), "context must be an object");
   ok(
-    context.user === undefined || typeof context.user?.id === "string",
-    "user.id must be a string if user is given",
+    typeof context.user === "undefined" || isObject(context.user),
+    "user must be an object if given",
+  );
+  ok(
```

Comment:
Same observation. We are repeating this code a lot so better to extract in functions.

Reply:
extracting